File: JULY-2025/14-07-2025/p1290.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class Solution {
  int getDecimalValue(ListNode? head) {
  ListNode? current=head;
  String str='';
  while(current!=null){
   str+=current.val.toString();
   current=current.next;
  }
  return int.parse(str,radix:2);
  }
}

File: JULY-2025/14-07-2025/p1290.kt
/**
 * Example:
 * var li = ListNode(5)
 * var v = li.`val`
 * Definition for singly-linked list.
 * class ListNode(var `val`: Int) {
 *     var next: ListNode? = null
 * }
 */
class Solution {
    fun getDecimalValue(head: ListNode?): Int {
        var binaryPresent = ""
        var head = head

        while (head != null) {
            binaryPresent += head.`val`.toString()
            head = head?.next
        }

        return binaryPresent.toInt(2)
    }
}